### PR TITLE
[Job Submission][refactor 4/N] Complete the remaining interfaces on JobAgent

### PR DESCRIPTION
**Pull request by Catch-Bull**
Signed-off-by: Catch-Bull <burglarralgrub@gmail.com>

## Why are these changes needed?

just need to implement stop_job, and I remove get_job_info because we can access JobInfoStorage without call `ray.init`.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by architkulkarni — APPROVED** [review]
Looks great! Could you say why we need both `agent_client` and `sdk_client` in the tests, and whether this is temporary?  It might be adding a brief code comment about this

File: dashboard/modules/job/job_head.py [hunk 1]
```diff
@@ -254,7 +258,6 @@ async def stop_job(self, req: Request) -> Response:
         )
 
     @routes.get("/api/jobs/{job_or_submission_id}")
-    @optional_utils.init_ray_and_catch_exceptions()
```
[the remark this answers is not in the document]
Comment (Catch-Bull) [reply dashboard/modules/job/job_head.py:257]: yeah, we can use `gcs_io_client` to access internal_kv and don't need to call `ray.init`

File: dashboard/optional_utils.py [hunk 2]
```diff
@@ -275,7 +275,7 @@ async def decorator(self, *args, **kwargs):
                     except Exception as e:
                         ray.shutdown()
                         raise e from None
-
+                    print("hejialing test:", self)
```
[the remark this answers is not in the document]
Comment (Catch-Bull) [reply dashboard/optional_utils.py:278]: thanks, I forget to remove it!

File: dashboard/modules/job/job_head.py [hunk 3]
```diff
@@ -76,23 +76,27 @@ async def submit_job_internal(self, req: JobSubmitRequest) -> JobSubmitResponse:
...
+
             if resp.status == 200:
                 result_json = await resp.json()
-                return JobDetails(**result_json)
+                return JobStopResponse(**result_json)
             else:
-                await self._raise_error(resp)
+                self._raise_error(resp)
 
-    async def get_job_logs(self, job_id: str) -> str:
```
[the remark this answers is not in the document]
Comment (Catch-Bull) [reply dashboard/modules/job/job_head.py:89]: thanks for commenting, It should be `get_job_logs_internal`, return response just to be consistent with `submit_job_internal` and `stop_job_internal`

**Pull request merged**